src/TestFramework/AbstractTestFrameworkAdapter.php

```diff
@@ -123,34 +124,19 @@ public function getMutantCommandLine(
         );
     }
 
+    /**
+     * @throws UnsupportedTestFrameworkVersion
+     */
     public function getVersion(): string
     {
-        if ($this->version !== null) {
-            return $this->version;
-        }
-
-        $testFrameworkVersionExecutable = $this->commandLineBuilder->build(
-            $this->testFrameworkExecutable,
-            [],
-            ['--version']
-        );
-
-        $process = new Process($testFrameworkVersionExecutable);
-        $process->mustRun();
-
-        $version = 'unknown';
-
-        try {
-            $version = $this->versionParser->parse($process->getOutput());
-        } catch (InvalidArgumentException $e) {
-            $version = 'unknown';
-        } finally {
-            $this->version = $version;
-        }
-
-        return $this->version;
+        return $this->version ?? $this->version = $this->retrieveVersion();
```

Comment:
Tiny refactoring to move the "retrieve code" in a clear place and just show clearly that we memoize the result here